src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1035,8 +1049,7 @@ public Collection<Object> createComponents(
             configPath,
             compatConfig
         );
-
-        final DynamicConfigFactory dcf = new DynamicConfigFactory(cr, settings, configPath, localClient, threadPool, cih);
+        dcf = new DynamicConfigFactory(cr, settings, configPath, localClient, threadPool, cih);
```

Comment:
Does `final` need to be removed here?

Reply:
Yes, I believe this is coming from the instantiate of CreateOnBehalfOfAction ([here](https://github.com/RyanL1997/security/commit/88f32e98304aa1aedfbcba6f471b74f6457262df#diff-e7ef66295cba81a49e4349781a2e9678d2b021a570e978bb4feb422b03d5d74aR535)). Since we do need the `dcf.registerDCFListener(cobot);`